lib/src/api/opt/resource.rs

```diff
@@ -1,58 +1,115 @@
-use crate::api::{err::Error, Result};
-use crate::sql::{self, Array, Edges, Id, Object, Table, Thing, Value};
-use crate::syn;
+use crate::{
+	api::{err::Error, Result},
+	Object, RecordId, RecordIdKey, Value,
+};
 use std::ops::{self, Bound};
+use surrealdb_core::sql::{
+	Edges as CoreEdges, IdRange as CoreIdRange, Table as CoreTable, Thing as CoreThing,
+};
+
+#[cfg(any(feature = "protocol-ws", feature = "protocol-http"))]
+use surrealdb_core::sql::Value as CoreValue;
+
+/// A wrapper type to assert that you ment to use a string as a table name.
+///
+/// To prevent some possible errors, by defauit [`IntoResource`] does not allow `:` in table names
+/// as this might be an indication that the user might have intended to use a record id instead.
+/// If you wrap your table name string in this tupe the [`IntoResource`] trait will accept any
```

Comment:
When is this needed?

Reply:
This was just the previous behavior but documented.

Reply:
Previously it was not possible to select from a table called `a:b` for example unless you went through the hassle of manually creating the `core::sql::Table` type.